test/distributed/test_data_parallel.py

```diff
@@ -597,6 +598,23 @@ def test_scatter_cpu(self):
     def test_scatter_gpu(self):
         self._test_scatter(torch.randn((4, 4)).cuda())
 
+    @unittest.skipIf(not TEST_MULTIGPU, "At least 2 CUDA GPUS needed")
+    @unittest.skipIf(NO_NCCL, "NCCL needed")
+    def test_data_parallel_complex(self):
+        class Cplx(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.cplx = torch.nn.Parameter(torch.zeros(1, 10, dtype=torch.cfloat).cuda())
```

Comment:
nit: Would be nice to add brief comments here + in the fwd function to explain what we expect (that complex tensors and viewed as reals, computed on, and then viewed back as complex transparently)